cpmpy/solvers/exact.py

```diff
@@ -0,0 +1,528 @@
+            elif my_status == 2: # found inconsistency
+                assert False, "Error: inconsistency during solveAll should not happen, please warn the developers of this bug"
+            elif my_status == 3: # found timeout
+                return solsfound
+
+        return solsfound
+
+    def has_objective(self):
+        """
+            Returns whether the solver has an objective function or not.
+        """
+        return self.objective_given
+
+
+    def solver_var(self, cpm_var, encoding="onehot"):
+        """
+            Creates solver variable for cpmpy variable
+            or returns from cache if previously created
+        """
+        if is_num(cpm_var):  # shortcut, eases posting constraints
+            return cpm_var
+
+        # variables to be translated should be positive
+        assert(not isinstance(cpm_var, NegBoolView))
+
+        # return it if it already exists
+        if cpm_var in self._varmap:
+            return self._varmap[cpm_var]
+
+        # create if it does not exist
+        revar = str(cpm_var)
+        if isinstance(cpm_var, _BoolVarImpl):
+            self.xct_solver.addVariable(revar,0,1)
+        elif isinstance(cpm_var, _IntVarImpl):
+            lb = cpm_var.lb
+            ub = cpm_var.ub
+            if max(abs(lb),abs(ub)) > 1e18:
+                # larger than 64 bit should be passed by string
+                self.xct_solver.addVariable(revar,str(lb), str(ub), encoding)
+            else:
+                self.xct_solver.addVariable(revar,lb,ub, encoding)
+        else:
+            raise NotImplementedError("Not a known var {}".format(cpm_var))
+        self._varmap[cpm_var] = revar
+        return revar
+
+
+    def objective(self, expr, minimize):
+        """
+            Post the given expression to the solver as objective to minimize/maximize
+
+            - expr: Expression, the CPMpy expression that represents the objective function
+            - minimize: Bool, whether it is a minimization problem (True) or maximization problem (False)
+
+            'objective()' can only be called once
+        """
+        if self.objective_given:
+            NotImplementedError("Exact accepts setting the objective function only once.")
+
+        self.objective_given = True
+        self.objective_minimize = minimize
+
+        # make objective function non-nested
+        (flat_obj, flat_cons) = flatten_objective(expr)
+        self += flat_cons  # add potentially created constraints
+        self.user_vars.update(get_variables(flat_obj))  # add objvars to vars
+
+        # make objective function or variable and post
+        xct_coefs,xct_vars,xct_rhs = self._make_numexpr(flat_obj,0)
+        if not self.objective_minimize:
+            xct_coefs = [-x for x in xct_coefs]
+
+        self.xct_solver.init(xct_coefs,xct_vars)
+        self.solver_is_initialized = True
+        self.xct_solver.setOption("verbosity","0")
+
+    def _make_numexpr(self, lhs, rhs):
```

Comment:
I would like this function to have same heading as other solver interfaces.
You can easily extract the lhs and rhs from the args of a comparison

Reply:
Ah although the return type is also different anyway... Ok maybe its fine to leave it as is then.

Reply:
ok